### DISCUSSION TITLE
ERR max number of clients + cluster connections reached when using Jedis Cluster

### DISCUSSION THREAD

**Discussion by bhargav-lime** [xetorthio/jedis#2797] [Q&A]
I am using Redis Cluster and with Jedis Client library,

As part of setup, we are currently using Redis Cluster and instantiating the cluster using the following mechanism.

```
JedisCluster(
    HostAndPort(
      System.getenv().getOrDefault("REDIS_SERVER_HOST", DEFAULT_HOST),
      System.getenv().getOrDefault("REDIS_SERVER_PORT", DEFAULT_PORT).toInt()
    )
  )
```
However, this code intermittently gives the following exception.

```
redis.clients.jedis.exceptions.JedisDataException: ERR max number of clients + cluster connections reached
```
I have looked at the [code reference](https://stackoverflow.com/questions/31070612/java-util-concurrent-executionexception-redis-clients-jedis-exceptions-jedisdat) here but it is not meaningful in my scenario since, I am not managing the connection manually and seems to be managed by Redis.

How can I resolve this error?

**Comment by sazzad16** [comment]
@bhargav-lime You would require to have some control to your Redis instances. Otherwise, it's almost impossible to solve the issue. If by "managed by Redis" you mean Redis Enterprise, contact them.

However, you can use
- [CLIENT LIST](https://redis.io/commands/client-list) command to list all the connected clients.
- [CLIENT KILL](https://redis.io/commands/client-kill) command to kill the clients of your choosing.

If you can't do any of these, you can restart your servers.

**Comment by zeekling** [comment]
@bhargav-lime Except control connection of redis instances, you can use config set maxclients 20000 to solve this issue in a short time.
Of course， the best solution is control connections.

There is another possibility， you not call `jedis.close()` method after you use the redis connection.
